### PR TITLE
round values before doing conditional release comparison

### PR DESCRIPTION
**Pull request by Alex-Jordan**
This addresses #1530. The discussion there includes a way to reproduce the issue.

### CODE REVIEW

File: lib/WeBWorK/Utils.pm [hunk 1]
```diff
@@ -1221,14 +1221,18 @@ sub is_restricted {
...
+		$restriction = sprintf("%.2f", $restriction);
 		my @good_sets;
 		foreach(@proposed_sets) {
 		  push @good_sets,$_ if $db->existsGlobalSet($_);
 		}
 		foreach(@good_sets) {
 	  	  my $restrictor =  $db->getGlobalSet($_);
 		  my $r_score = grade_set($db,$restrictor,$_, $studentName,0);
+		  # round to evade machine rounding error
+		  $r_score = sprintf("%.2f", "$r_score");
```
Comment (drgrice1) [inline lib/WeBWorK/Utils.pm:1235]: There is a minor issue here.  The `$r_score` inside the `sprintf` should not be quoted.  That doesn't do anything useful and merely adds the inefficiency of perl needing to interpolate the string.

**Comment by Alex-Jordan** [comment]
I removed those quotes.

BTW, at first I tried doing the rounding in the same line that initiates `$r_score`:

```
 my $r_score = sprintf("%.2f", grade_set($db,$restrictor,$_, $studentName,0));
```
I had an example where the student score was 0.75. But apache logging was showing that `$r_score` was coming out as 7.5 instead with the above line of code. I cannot explain this. 

I reconfigured it over two lines and that appears to work as intended. (The logs showed 0.75.) Those quotes were in there as part of my probing to understand how 7.5 could come out and I forgot to remove them.

**Comment by drgrice1** [comment]
The reason that the single line code
```
my $r_score = sprintf("%.2f", grade_set($db,$restrictor,$_, $studentName,0));
```
did not work is because as when you put it as the second argument (or really the list of remaining arguments) for sprintf it is in list context.  The grade_set returns a list with first element the total right and the second element the total when it is called in list context.  If you instead use
```
my $r_score = sprintf("%.2f", scalar(grade_set($db,$restrictor,$_, $studentName,0)));
```
it would work.

**Comment by drgrice1** [comment]
Note that `my $restriction` could also be directly assigned the sprintf value.  There you don't have to worry about the `scalar` usage as it is directly a scalar to begin with (no wantarray usage).

**Pull request merged**